WordPress/wordcamp.org
# Speaker Feedback: Add REST endpoint for updating feedback with specific meta values

Adds the ability to mark a feedback comment as "helpful" via a new API endpoint. This is stored as comment meta, just like the other values that make up the original feedback submission. However, "helpful" can only be set during feedback update, not creation. This is achieved by introducing a "context" for each possible meta field.

This also makes some changes to the way request objects are processed by our endpoint controller to allow for the flexibility to consider the context of the request.

Refs #342 

### How to test the changes in this Pull Request:

1. To make is easier to send authenticated requests to the endpoint via something like [Insomnia](https://insomnia.rest/), you'll probably want to temporarily install the [Basic Auth plugin](https://github.com/WP-API/Basic-Auth).
1. Create some feedback comments on one or more session posts.
1. To mark one of the comments as helpful, you'll need the ID of the comment.
1. The endpoint URL will be `[your-test-site].wordcamp.org/wp-json/wordcamp-speaker-feedback/v1/feedback/[comment ID]`
1. Currently there is only valid parameter for the payload:
    * `meta[helpful]` (A boolean-ish value)
1. The response to a successful request to the endpoint should be a 201 status and an empty array.
1. Try sending a non-boolean value. You should get an error message.
1. Try including other existing meta values besides `helpful` (`rating` or `q1` for example). These should be ignored and not changed during a successful request.
1. Try including `meta[helpful]` when sending a request to [create a new feedback](https://github.com/WordPress/wordcamp.org/pull/353). It should also be ignored and not be present for the newly created comment.
1. Try authenticating as a non-admin/editor user who is also not a speaker on the session. You should get an error message.
